### PR TITLE
[RTM] Added the insert tag flag for absolute URLs

### PR DESCRIPTION
**Pull request by qzminski**
Allows for `{{news_url::123|absolute}}`. Related to contao/core-bundle#970.

### CODE REVIEW

File: src/EventListener/InsertTagsListener.php [hunk 1]
```diff
@@ -48,10 +48,13 @@ public function __construct(ContaoFrameworkInterface $framework)
...
      *
      * @param string $tag
+     * @param bool   $useCache
+     * @param mixed  $cacheValue
+     * @param array  $flags
      *
      * @return string|false
      */
-    public function onReplaceInsertTags(string $tag)
+    public function onReplaceInsertTags(string $tag, bool $useCache, $cacheValue, array $flags)
```
Comment (leofeyer) [inline src/EventListener/InsertTagsListener.php]: This cannot be right. First, both `$useCache` and `$cacheValue` are not used anywhere, so why did you add them? Second, if you add new arguments to an existing method, you should make them optional. If you cannot make them optional (which is not the case here), you have to adjust the `listeners.yml` file accordingly.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: src/EventListener/InsertTagsListener.php (hunk 1 above)
Comment (qzminski) [reply src/EventListener/InsertTagsListener.php]: I have added the `$useCache` and `$cacheValue` to get the access to the `$flags`. Should I use `func_get_arg()` instead?

Comment (qzminski) [reply src/EventListener/InsertTagsListener.php]: The `core-bundle` passes the arguments in that order: https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/InsertTags.php#L953

Comment (leofeyer) [reply src/EventListener/InsertTagsListener.php]: Ah, now I understand. Then its fine I guess.

Comment (qzminski) [reply src/EventListener/InsertTagsListener.php]: But the new arguments should be optional indeed or otherwise it is a BC break – even if this method is used only by the hook. Should I change that?

Comment (leofeyer) [reply src/EventListener/InsertTagsListener.php]: Yes, I think using `func_get_arg()` is the best solution in this case.

Comment (qzminski) [reply src/EventListener/InsertTagsListener.php]: Sorry, by optional I meant providing their default values:

```php
public function onReplaceInsertTags(string $tag, bool $useCache = true, $cacheValue = null, array $flags = [])
```

Comment (aschempp) [reply src/EventListener/InsertTagsListener.php]: Symfony does a lot of `func_num_args() > 4` … check to see if the argument is there and then use it.

Comment (qzminski) [reply src/EventListener/InsertTagsListener.php]: How about this? 599ea6d

Comment (leofeyer) [reply src/EventListener/InsertTagsListener.php]: No, Symfony only does this for deprecated stuff, e.g.

https://github.com/symfony/symfony/blob/8cd2193a82171686f41fd073d2157de0293e3bb2/src/Symfony/Component/Yaml/Yaml.php#L93-L107

But never for regular method calls.

Comment (qzminski) [reply src/EventListener/InsertTagsListener.php]: So let's use the "normal" approach in 4fb9357.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @qzminski.
